### PR TITLE
Tumbleweed icons&colors changes

### PR DESCRIPTION
**Pull request by ilvipero**
1. air-openSUSE. Left it as it was, using oxygen icons. Very good theme. we might want to switch to the icons i customized for openSUSElight, darker icons might look better with air-openSUSE. but can do that later.
2. openSUSE. I changed it to have a green start icon, dark bars with green accents, oxygen icons fallback, left transitions as they were in openSUSElight and openSUSEdark.
3. openSUSElight. I removed the icons as the icon pack was not complete. mixed with breeze did not have a very good feel to it, in my opinion. So i modified all oxygen icons, inverted the colors so that it suits well on a light theme. (I will also submit oxygen-light icons to KDE and see if they like it).
4. openSUSEdark. I removed the icons as the icon pack was not complete. mixed with breeze did not have a very good feel to it, in my opinion. replaced with oxygen icons fallback.

### CODE REVIEW

**Comment by Vogtinator** [comment]
Changing the start icon would lose the adaption to color scheme changes, so I'm against that.

In the theme metadata, the `X-KDE-PluginInfo-Version` needs to be bumped as otherwise the cache is not updated.

**Comment by Vogtinator** [comment]
The openSUSE theme needs to be close to upstream, so it has to use Breeze as fallback (which is the default).

**Comment by ilvipero** [comment]
Regarding green icon on opensuse theme: Sure, I thought to change it a bit to avoid the theme being too similar to openSUSEdark. But if you are not for it, I can revert back.

Regarding fallback to breeze: Ok. How about openSUSEdark and openSUSElight themes? Should those also fallback to breeze dark and breeze light?

**Comment by lbeltrame** [comment]
Some screenshots would be nice.

**Comment by ilvipero** [comment]
> Some screenshots would be nice.

Sure, Please check attached screenshots. I am also attaching all 3 themes packaged. I renamed  both the archives and theme names in metadata files, so that you guys can import manually and they won't conflict with existing opensuse themes.
I hope it helps.

opensuse-theme-screenshot
opensusedark-theme-screenshot
opensuselight-theme-screenshot
[openSUSE-workinprogress.tar.gz](https://github.com/openSUSE/plasma-opensuse/files/3139037/openSUSE-workinprogress.tar.gz)
[openSUSEdark-workinprogress.tar.gz](https://github.com/openSUSE/plasma-opensuse/files/3139040/openSUSEdark-workinprogress.tar.gz)
[openSUSElight-workinprogress.tar.gz](https://github.com/openSUSE/plasma-opensuse/files/3139048/openSUSElight-workinprogress.tar.gz)

**Comment by Vogtinator** [comment]
> Regarding fallback to breeze: Ok. How about openSUSEdark and openSUSElight themes? Should those also fallback to breeze dark and breeze light?

I'd say yes, unless it looks bad.

**Comment by ilvipero** [comment]
> > Regarding fallback to breeze: Ok. How about openSUSEdark and openSUSElight themes? Should those also fallback to breeze dark and breeze light?
> 
> I'd say yes, unless it looks bad.

No, it does not look bad at all. I am going to make the changes.

**Comment by ilvipero** [comment]
> Changing the start icon would lose the adaption to color scheme changes, so I'm against that.

I tried looking at the manual to see how I can keep same icon and have it adapt to color schemes, but did not find which configuration does that. Could you point me in the right direction? Happy to learn and change per your comment.

**Comment by Vogtinator** [comment]
> No, it does not look bad at all. I am going to make the changes.

Thanks!

> > Changing the start icon would lose the adaption to color scheme changes, so I'm against that.
> 
> I tried looking at the manual to see how I can keep same icon and have it adapt to color schemes, but did not find which configuration does that. Could you point me in the right direction? Happy to learn and change per your comment.

You need to specify special CSS style sheets with a special id in the .svg file: https://techbase.kde.org/Development/Tutorials/Plasma5/ThemeDetails#Using_system_colors

**Comment by ilvipero** [comment]
I completed all the changes per your request. I decided to leave the original start icon on all themes, it looks quite good to me. 
Could you please check and let me know if further changes are needed? 

Thanks!

**Comment by Vogtinator** [comment]
Can you do another round of screenshots please?

**Comment by ilvipero** [comment]
openSUSE Dark Theme. This one had mismatching icons and some missing ones. I removed all the custom icons, kept animations, colors as they were. and defaulted to breeze-dark.

opensuse-dark-screenshot01

**Comment by ilvipero** [comment]
openSUSE Light Theme. This one was fine but icons were not upstream and looked a bit too different from other themes. I removed all the custom icons, kept animations, colors as they were. and defaulted to breeze-light.

opensuse-light-screenshot01

**Comment by ilvipero** [comment]
openSUSE Theme. I made this theme a bit more green. defaulted icons to breeze-dark.
I would have liked a green start logo to match the green openSUSE theme. But dynamically assigned colors did not look good. So for now i reverted back to the black and white icon. 

opensuse-screenshot01

The icons below would look good on this theme, at least in my humble opinion. What do you think? do you guys have something green we can use as the main openSUSE theme?

openSUSE-start

**Comment by Vogtinator** [comment]
I'd like to have input from @hellcp here as well.

Do you have any .svgz copied from the theme set as fallback here? If so, those could be deleted. AFAICT the diff between the upstream theme set as fallback and the downstream result could be much smaller.

Also note that this seems to conflict with #14 by @guoyunhe, so you need to coordinate the changes.

**Comment by ilvipero** [comment]
I removed all icons that come from breeze theme, and set breeze as fallback. so that the theme is smaller.

Regarding conflict, that seems related to background colors, I am happy to work with @guoyunhe

**Comment by guoyunhe** [comment]
Hi @ilvipero , my PR is only for the color change and it is small. You can download the patch and merge it into yours (probably modify if here are things wrong in my patch). Then I can close my PR and only merge this one. Cheers.

My patches:

[patches.zip](https://github.com/openSUSE/plasma-opensuse/files/3381189/patches.zip)

**Comment by ilvipero** [comment]
@guoyunhe Thank you, I am happy to merge your patch in my pull request. 

謝謝你 :)

**Comment by ilvipero** [comment]
Sorry for slow reply. It seems my last comment was gone. I completed all changes from request #14 a while back and my changes are done.

**Comment by DanielNoga** [comment]
I like that icons are removed. For example I didn't know until now, why my VPN plasma-nm icon was almost invisible: https://bugzilla.opensuse.org/show_bug.cgi?id=1134234

Can it be backported to Leap 15.1 after merge?

**Comment by ilvipero** [comment]
@Vogtinator could you please check and if no further changes are needed, merge the request?
Thanks!

**Comment by ilvipero** [comment]
@DanielNoga this is the bug I opened for the same issue. I will mark it resolved if the pull requests is merged. https://bugzilla.opensuse.org/show_bug.cgi?id=1128854

**Comment by Vogtinator** [comment]
> Can it be backported to Leap 15.1 after merge?

This whole PR definitely not, it's too big and invasive (it changes the default theme!). For 15.2 it's doable though.

File: config-files/usr/share/plasma/desktoptheme/openSUSElight/metadata.desktop
Comment (Vogtinator) [inline config-files/usr/share/plasma/desktoptheme/openSUSElight/metadata.desktop:3]: `Light`

File: config-files/usr/share/plasma/desktoptheme/openSUSE/lancelot/theme.config
Comment (Vogtinator) [inline config-files/usr/share/plasma/desktoptheme/openSUSE/lancelot/theme.config]: AFAIK lancelot is dead, so this could be removed.

**Review by Vogtinator — CHANGES_REQUESTED** [review]
So you removed many icons, but also added some new .svgz files. Where are they from? IMO the default openSUSE theme should use upstream icons wherever possible.

If they're just copied from breeze/oxygen/etc., please let the theme inherit them instead.

File: config-files/usr/share/plasma/desktoptheme/openSUSElight/metadata.desktop
Comment (ilvipero) [reply config-files/usr/share/plasma/desktoptheme/openSUSElight/metadata.desktop]: Yes you are right. I removed all icons so that themes can inherit from upstream (breeze-light and breeze-dark). 
The svgz I added back:
opensuse start.sgvz , in the icons folder. 
indicators.svgz , in the icontasks folder. This one was already in opensuse themes, it looks good so I believe we can keep it. If you prefer to remove it, please let me know.

start-svgz
indicators-svgz

File: config-files/usr/share/plasma/desktoptheme/openSUSE/lancelot/theme.config
Comment (ilvipero) [reply config-files/usr/share/plasma/desktoptheme/openSUSE/lancelot/theme.config]: Shall I remove the whole lancelot folder and its contents from all themes?

**Comment by ilvipero** [comment]
> So you removed many icons, but also added some new .svgz files. Where are they from? IMO the default openSUSE theme should use upstream icons wherever possible.
> 
> If they're just copied from breeze/oxygen/etc., please let the theme inherit them instead.

Yes I removed most icons and only left the default opensuse start.svgz and indicators.svgz
indicators.svgz was previously in opensuse themes, I did not create it, just kept it. it looks good to me and we can keep it in my opinion. But if you want the whole theme to be upstream breeze, I can remove it as well.

start-svgz
indicators-svgz

**Comment by noahadvs** [comment]
One thing I would like to see either now or in the future is the use of SVG in the repo instead of SVGZ. Upstream Breeze switched to using SVG in the repo and Git isn't able to read the contents of SVGZ files, so they cause lots of repo history bloat and make patch review more difficult. When the upstream Breeze desktop theme is built, the SVG files are converted to SVGZ.

File: config-files/usr/share/plasma/desktoptheme/openSUSE/lancelot/theme.config
Comment (Vogtinator) [reply config-files/usr/share/plasma/desktoptheme/openSUSE/lancelot/theme.config]: Instead of adding files, yes. Keeping the old stuff around is IMO fine.

File: config-files/usr/share/plasma/desktoptheme/openSUSE/widgets/branding.svgz
Comment (Vogtinator) [inline config-files/usr/share/plasma/desktoptheme/openSUSE/widgets/branding.svgz]: Is a plain SVG file, but with `svgz` extension.

**Review by Vogtinator — CHANGES_REQUESTED** [review]
> > So you removed many icons, but also added some new .svgz files. Where are they from? IMO the default openSUSE theme should use upstream icons wherever possible.
> > If they're just copied from breeze/oxygen/etc., please let the theme inherit them instead.
> 
> Yes I removed most icons and only left the default opensuse start.svgz and indicators.svgz
> indicators.svgz was previously in opensuse themes, I did not create it, just kept it. it looks good to me and we can keep it in my opinion. But if you want the whole theme to be upstream breeze, I can remove it as well.

Currently there are still a lot of added svgs for the openSUSE theme, which should be as minimal as possible to make it viable as a default theme.
Similar for openSUSE-light and -dark, is it possible to minimize the deviation from breeze?
If the upstream icons were not fitting/usable or even missing, can you submit those upstream instead?

Can you change the `.svgz` you added to `.svg` as @noahadvs requested? That way there are no renames and those can be compared directly.

File: config-files/usr/share/plasma/desktoptheme/openSUSE/widgets/branding.svgz
Comment (ilvipero) [reply config-files/usr/share/plasma/desktoptheme/openSUSE/widgets/branding.svgz]: The file was renamed per your request.

Comment (ilvipero) [reply config-files/usr/share/plasma/desktoptheme/openSUSE/widgets/branding.svgz]: So, originally I did not want to deviate too much from the themes that are currently on opensuse tumbleweed. Most of them are heavily customized compared to breeze variants. 
I thought to just tweak the minimum to get it improved. But if you prefer to have a major overhaul of those and bring opensuse themes more in line with breeze, I am happy to help and will do my best.

Comment (Vogtinator) [reply config-files/usr/share/plasma/desktoptheme/openSUSE/widgets/branding.svgz]: > I thought to just tweak the minimum to get it improved. But if you prefer to have a major overhaul of those and bring opensuse themes more in line with breeze, I am happy to help and will do my best.

If the theme is meant to be enabled by default, this is IMO a necessary precondition. For non-default themes it doesn't matter as much.

Comment (ilvipero) [reply config-files/usr/share/plasma/desktoptheme/openSUSE/widgets/branding.svgz]: I have no issue at all with that. I did not create those themes so I did not want to butcher other people's work too much :)
So if I remember well we already fallback to breeze dark or breeze light on those 3 themes. Means if no item is found, fallback theme is used instead. Does it mean we can basically remove most of the icons and widgets and only leave the branding and start icons?

Comment (Vogtinator) [reply config-files/usr/share/plasma/desktoptheme/openSUSE/widgets/branding.svgz]: The branding and start icons are already injected into the breeze and oxygen icon themes (`/usr/share/icons/breeze/places/16/start-here-branding.svg`) so not part of the theme or lnf.

Comment (hellcp) [reply config-files/usr/share/plasma/desktoptheme/openSUSE/widgets/branding.svgz]: Maybe we could also adjust the highlight/accent colours, considering Plasma guys want to make that easier for the users anyway https://phabricator.kde.org/T11052

Comment (hellcp) [reply config-files/usr/share/plasma/desktoptheme/openSUSE/widgets/branding.svgz]: And I mean breeze green, not openSUSE green, openSUSE green really doesn't work well with breeze

Comment (ilvipero) [reply config-files/usr/share/plasma/desktoptheme/openSUSE/widgets/branding.svgz]: @Vogtinator @hellcp  I made all the modifications, left a very small set of files for color schemes, metadata and start icon in svg format. I will upload a few screenshots and from those you can also see the themes folder structure is very clean. Please let me know if it looks ok.
@guoyunhe your patch was not removed. Your color changes still apply.

Comment (ilvipero) [reply config-files/usr/share/plasma/desktoptheme/openSUSE/widgets/branding.svgz]: openSUSE theme.
openSUSE-breezedefaults

openSUSE dark theme.
openSUSEdark-breezedefaults

openSUSE light theme.
openSUSElight-breezedefaults

**Comment by Vogtinator** [comment]
Looking at the diff (not the commits) there are still many `.svg` files getting added, which is a maintenance burden.
Except for openSUSE-specific SVGs, everything else has to be inherited from the upstream theme.
The old themes didn't do that, which is the reason why they're unmaintained/incomplete now.

IMO though `openSUSE` should stay as a light theme as it is right now, but we can switch the default desktoptheme to `openSUSEdark`. Otherwise users of the default `openSUSE` theme would suddenly have a dark shell, which isn't really expected.

Theme  | Inherits from | Changes to the inherited theme
------------ | ------------- | ---
openSUSE | Breeze | `start.svgz`, `indicators.svgz` and maybe colors (?)
openSUSE Dark | Breeze Dark | `start.svgz`, `indicators.svgz` and colors
openSUSE Light | Breeze Light | `start.svgz`, `indicators.svgz` and colors (?)

The screenshots look good and I'm also in favor of switching to a dark theme by default, but the majority of `.svg` files here are AFAICT not necessary and should be deleted. Essentially, what should be done is to create a new theme folders from scratch and only copy over necessary files.

**Comment by ilvipero** [comment]
@Vogtinator Thanks for the insight. I will make brand new clean folders and just leave the very few files for color scheme and start.svgz.

**Comment by ilvipero** [comment]
@Vogtinator 
Per you recommendation, I cleaned up unnecessary .svg(z) to bring themes close to upstream.
FYI, there is one theme, openSUSEdefault. It takes the fallback from the air theme, not breeze. I did not know what you wanted to do with it, so I left it there.

**Comment by DanielNoga** [comment]
Hello, what is status here? It would be good to fix missing icons in dark theme by merging this patch

**Comment by ilvipero** [comment]
> Hello, what is status here? It would be good to fix missing icons in dark theme by merging this patch

Hi, I have done all the changes that were asked, a long time ago, and was waiting for more feedback or confirmation it could be merged. @Vogtinator

**Comment by Vogtinator** [comment]
I'll merge this now (squashing everything) for the Plasma 5.24 submission to Tumbleweed, and fix some minor issues I noticed while testing this.

However, I'll revert some color changes for now:

Switching the default to a dark theme is a rather invasive change, as it hits existing installations. On top of the change itself, it appears like some Plasma caches aren't invalidated and after installation there's an unnusable mixup of old and new colors being used, leading to e.g. unreadable text. (It might be possible to address some of this by switching the LNF to use `openSUSEdark`, instead of making `openSUSE` a dark theme, but I haven't tried yet)

The openSUSE color scheme results in green folder icons, and the contrast between the green background and the symbols on top is rather poor:

Screenshot_20220116_215519

We can revert the reverts once the issues got addressed.

**Pull request merged**
